<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />

  <!-- set to false if the nodelet should run under an external nodelet manager -->
  <arg name="standalone" default="true" />

  <!-- allows disabling the gui from command line -->
  <arg name="gui" default="true" />

  <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="manager" default="$(arg UAV_NAME)_masters_manager" />
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>

  <!-- Namespace - all topics and parameters will be remapped using this namespace as a prefix (eg.: "/vision_example/image_in" to "/uav1/vision_example/image_in") -->
  <group ns="$(arg UAV_NAME)">
    <node pkg="tf" type="static_transform_publisher" name="rtk_connector" args="0 0 0 0 0 0 1 uav27/rtk_origin uav28/rtk_origin 100" />


    <!-- Masters nodelet -->
    <node pkg="nodelet" type="nodelet" name="masters" args="$(arg nodelet) masters/Masters $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Parameters loaded from launch file -->
      <param name="gui" type="bool" value="$(arg gui)"/>

      <param name="UAV_NAME" type="string" value="$(arg UAV_NAME)"/>
      <param name="world_origin" type="string" value="$(arg UAV_NAME)/world_origin"/>

      <param name="eagle_name" type="string" value="$(arg UAV_NAME)"/>
      <param name="target_name" type="string" value="uav2"/>

<!--  For the rosbags recorded before -->
<!--       <param name="eagle_gt_odometry" type="string" value="/$(arg UAV_NAME)/estimation_manager/ground_truth/odom"/> -->
<!--       <param name="target_gt_odometry" type="string" value="/uav2/estimation_manager/ground_truth/odom"/> -->
<!--       <param name="main_camera" type="string" value="/$(arg UAV_NAME)/bluefox_down"/> -->

<!--  For the simulation -->
<!--       <param name="eagle_gt_odometry" type="string" value="/$(arg UAV_NAME)/estimation_manager/ground_truth/odom"/> -->
<!--       <param name="target_gt_odometry" type="string" value="/uav2/estimation_manager/ground_truth/odom"/> -->
<!--       <param name="main_camera" type="string" value="/$(arg UAV_NAME)/rgbd_front/color"/> -->

<!--  For the provided rosbag and real world (I guess) -->
<!--       <param name="lidar_tracker" type="string" value="/$(arg UAV_NAME)/lidar_tracker/tracks"/> -->
<!--       <param name="eagle_gt_odometry" type="string" value="/$(arg UAV_NAME)/estimation_manager/ground_truth/odom"/> -->
<!--       <param name="target_gt_odometry" type="string" value="/uav2/estimation_manager/ground_truth/odom"/> -->
<!--       <param name="main_camera" type="string" value="/$(arg UAV_NAME)/rgbd_pitched/color"/> -->

<!--  For the rosbag from the camp (NO lidar) -->
      <param name="eagle_gt_odometry" type="string" value="/uav27/estimation_manager/rtk/odom"/>
      <param name="target_gt_odometry" type="string" value="/uav28/estimation_manager/odom_main"/>
      <param name="target_tf_name" type="string" value="/uav28/estimation_manager/odom_main"/>
      <param name="main_camera" type="string" value="/uav27/up_rgbd/color/"/>


      <!-- ROS parameters config file -->
      <rosparam file="$(find masters)/config/default.yaml" />
      <rosparam file="$(find masters)/config/masters.yaml" />

      <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->

      <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->

    </node>

  </group>

</launch>
